### PR TITLE
add query pod list feature

### PR DESCRIPTION
**Pull request by zhangtong007**
discription:
add api: /apis/v1/debugpodlist. Used to query the pod list for diagnosis based on poduid, podname, podip, and pod hostname
test：
Access test through api：
image

Access test through grafana：
image

### CODE REVIEW

**Review by d3c3mber — COMMENTED** [review]
Can you please add some description and a few use cases to this PR, thanks

**Comment by zhangtong007** [comment]
> Can you please add some description and a few use cases to this PR, thanks

discription and test have been added

File: internal/grafanadi/handler/debugpodlist.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+	begin := time.Now()
+	defer func() {
+		cost := utils.TimeSinceInMilliSeconds(begin)
+		metrics.QueryMethodDurationMilliSeconds.WithLabelValues("QueryDebugPodListWithPodUid").Observe(cost)
+	}()
+
+	var err error
+	if key == "uid" {
+		err = handler.storage.QueryPodYamlsWithPodUID(&podYamls, value)
```
Comment (larryck) [inline internal/grafanadi/handler/debugpodlist.go:69]: I think the QueryPodYamlsWithPodUID func  returns only one pod, does that match your purpose?

File: internal/grafanadi/handler/debugpodlist.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"time"
+
+	"github.com/alipay/container-observability-service/internal/grafanadi/service"
+	"github.com/alipay/container-observability-service/pkg/dal/storage-client/data_access"
+	"github.com/alipay/container-observability-service/pkg/dal/storage-client/model"
+	"github.com/alipay/container-observability-service/pkg/metrics"
+	"github.com/alipay/container-observability-service/pkg/utils"
+)
+
+type DebugPodListHandler struct {
```
Comment (larryck) [inline internal/grafanadi/handler/debugpodlist.go:15]: The word DebugPodList is misleading, QueryPodList maybe better.

File: internal/grafanadi/service/debugpodlist.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/alipay/container-observability-service/internal/grafanadi/model"
+	storagemodel "github.com/alipay/container-observability-service/pkg/dal/storage-client/model"
+)
+
+func ConvertPodyamls2Table(podyamls []*storagemodel.PodYaml) []*model.DebugPodListTable {
+	podyamlsSet := DeduplicateAndTopK(podyamls, 5)
+
+	tables := make([]*model.DebugPodListTable, 0)
+	for _, v := range podyamlsSet {
+		var state string = "在线"
```
Comment (d3c3mber) [inline internal/grafanadi/service/debugpodlist.go:16]: can you please use keyword in english for the state and please keep the naming consistent across functionalities, thanks :D

File: internal/grafanadi/handler/debugpodlist.go (hunk 1 above)
Comment (zhangtong007) [reply internal/grafanadi/handler/debugpodlist.go:69]: fixed

File: internal/grafanadi/handler/debugpodlist.go (hunk 2 above)
Comment (zhangtong007) [reply internal/grafanadi/handler/debugpodlist.go]: fixed

File: internal/grafanadi/service/debugpodlist.go (hunk 3 above)
Comment (zhangtong007) [reply internal/grafanadi/service/debugpodlist.go]: fixed

File: internal/grafanadi/server/server.go [hunk 4]
```diff
@@ -67,6 +67,9 @@ func (s *Server) StartServer(stopCh chan struct{}) {
 		r.Path("/elasticaggregations").HandlerFunc(corsWrapper(interutils.ServeSLOGrafanaDI, s.Storage))
 		r.Path("/rawdata").HandlerFunc(handlerWrapper(handler.RawdataFactory, s.Storage))
 
+		// federation api
+		r.Path("/apis/v1/debugpodlist").HandlerFunc(handlerWrapper(handler.QueryPodListFactory, s.Storage))
```
Comment (larryck) [inline internal/grafanadi/server/server.go:71]: what about rename debugpodlist api to  querypodlist?

**Review by d3c3mber — APPROVED** [review]

**Review by linuzb — APPROVED** [review]
LGTM

**Pull request merged**